core/src/main/java/org/hibernate/ogm/dialect/query/spi/RecognizerBasedParameterMetadataBuilder.java

```diff
@@ -24,32 +23,13 @@
 
 	@Override
 	public ParameterMetadataImpl buildParameterMetadata(String nativeQuery) {
-		ParamLocationRecognizer recognizer = new ParamLocationRecognizer();
+		ParamLocationRecognizer recognizer = new ParamLocationRecognizer( 0 );
```

Comment:
So... this is the change in ORM 5.3 that I find the most concerning (the fact that the base for parameter numbering has changed). I'm really worried about it, even if we have a configuration knob to restore the old behavior.

I understand you chose to be entirely compatible with the old behavior of OGM? I must admit that I'm wondering if we should follow the ORM way or be more conservative. I'm a bit torn on this subject.

@Sanne what's your take on this?

Reply:
To be fair, this is one of the part where I need to investigate more, they now have compliance settings that should help use to decide. For now, it seemed easier to be conservative.

Reply:
Yes, that's what I suspected. But TBH, I'm unsure on how we should follow up on that. It seems annoying to break every applications using ordinal parameters by default. But on the other hand, I wouldn't want us to be inconsistent with ORM. I can see heated discussions coming :)

Reply:
that's another one of the annoying "JPA mandates it" cases... we'll just need to follow the example of ORM.